### DISCUSSION TITLE
Recursively resolve URIs

### DISCUSSION THREAD

**Discussion by 0pq76r** [openssl/openssl#23590] [Q&A] [answered]
I have a question regarding the provider-object type OSSL_OBJECT_NAME and recursively loading URIs.

> OSSL_OBJECT_NAME
> The object data may only be passed by value, and should be a UTF8 string.
>
>This is useful for [provider-storemgmt(7)](https://www.openssl.org/docs/manmaster/man7/provider-storemgmt.html) when a URI load >results in new URIs.
-- [provider-object](https://www.openssl.org/docs/manmaster/man7/provider-object.html)

The documentation seems to indicate that the OSSL_OBJECT_NAME can be used by the store implementation in a provider to return further URIs. 

However, I cannot get OpenSSL to process the returned URIs. It only works for the `storeutl` command when enabling the recursive flag. There the recursive iteration is implemented in the application.
https://github.com/openssl/openssl/blob/4f6133f9db2b9b7ce5e59d8b8ec38202a154c524/apps/storeutl.c#L437-L443

Is possible to get the recursive loading of URIs working in other places or is it intended that the application has to do the recursive calls?

**Comment by t8m** [comment]
Perhaps @levitte might want to answer this?

**Comment by levitte** [comment, marked as answer]
Getting NAME answers from OSSL_STORE means you gave it a URI that represents a catalogue of resources. Essentially, it acts like opendir() / readdir() do with directories.

OSSL_STORE can't know what the application wants to do with the returned names, so reusing is indeed left to the application.

**Reply by 0pq76r** [reply]
Not the answer I was hoping for, but I guess it makes sense. Thanks for the quick response.
